### PR TITLE
Fix #934: Add support for proximity anti-fraud check to offline mode

### LINKED ISSUE #934: Add support for proximity anti-fraud check to offline mode
A follow-up to #905

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/OfflineSignatureServiceBehavior.java [hunk 1]
```diff
@@ -336,6 +336,31 @@ private VerifyOfflineSignatureResponse verifyOfflineSignatureImpl(String activat
...
+            return true;
+        }
+
+        final int steps = request.getProximityCheckStepCount();
+        logger.debug("Verifying proximity TOTP, activation ID: {}, steps count: {}", request.getActivationId(), steps);
+
+        final String[] dataElements = request.getDataString().split("&");
+        final String operationData = dataElements[dataElements.length - 1];
+
+        final String[] operationDataElements = new String(Base64.getDecoder().decode(operationData), StandardCharsets.UTF_8).split("&");
```
[the remark this answers is not in the document]
Comment (petrdvorak) [reply powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/OfflineSignatureServiceBehavior.java:351]: Maybe just extracting `Base64.getDecoder().decode(operationData)` would make this better.

**Review by petrdvorak — APPROVED** [review]
Looks OK. Just double-check that the auto-wiring changes have no unintended side effects.

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/OfflineSignatureServiceBehavior.java [hunk 2]
```diff
@@ -336,10 +338,10 @@ private VerifyOfflineSignatureResponse verifyOfflineSignatureImpl(final VerifyOf
         }
     }
 
-    private boolean processProximityCheck(final VerifyOfflineSignatureParameter request, final int digitsNumber) throws CryptoProviderException {
+    private VerifyOfflineSignatureResponse.ProximityCheck processProximityCheck(final VerifyOfflineSignatureParameter request, final int digitsNumber) throws CryptoProviderException {
         if (StringUtils.isBlank(request.getProximityCheckSeed())) {
             logger.debug("Proximity seed is not present and is TOTP not being verified, activation ID: {}", request.getActivationId());
-            return true;
+            return null;
```
Comment (petrdvorak) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/OfflineSignatureServiceBehavior.java:344]: I would add a comment here, just to emphasize the difference between returning `proximityCheck.success=true`.

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/CreatePersonalizedOfflineSignaturePayloadParameter.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import java.time.Duration;
+
+/**
+ * Parameter object for {@link OfflineSignatureServiceBehavior#createPersonalizedOfflineSignaturePayload(CreatePersonalizedOfflineSignaturePayloadParameter)}.
+ *
+ * @author Lubos Racansky, lubos.racansky@wultra.com
+ */
+@Builder
+@Getter
+public class CreatePersonalizedOfflineSignaturePayloadParameter {
```
Comment (romanstrobl) [inline powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/CreatePersonalizedOfflineSignaturePayloadParameter.java:34]: Is such long class name required?

**Review by romanstrobl — APPROVED** [review]
I have just one minor nit-pick related to class naming conventions, otherwise great job!

File: powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/CreatePersonalizedOfflineSignaturePayloadParameter.java (hunk 3 above)
Comment (banterCZ) [reply powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/CreatePersonalizedOfflineSignaturePayloadParameter.java]: Ok, let's rename it to `OfflineSignatureParameter`.

**Pull request merged**